Delete copy of quantized SDPA in torchao/experimental

metascroy opened the pull request:
Summary:
This deletes the SDPA/matmul kernels in torchao/experimental.

These kernels were already migrated to torchao/csrc/cpu/torch_free_kernels in the first diff, but we needed to keep them around in torchao/experimental to do a pin bump in ExecuTorch OSS after the first diff lands.

Differential Revision: D81640227

kimishpatel reviewed (approved):
Review automatically exported from Phabricator review in Meta.

metascroy committed:
Delete copy of quantized SDPA in torchao/experimental (#13924)

Summary:
X-link: https://github.com/pytorch/ao/pull/2952

This deletes the SDPA/matmul kernels in torchao/experimental.

These kernels were already migrated to torchao/csrc/cpu/torch_free_kernels in the first diff, but we needed to keep them around in torchao/experimental to do a pin bump in ExecuTorch OSS after the first diff lands.

Pull Request resolved: https://github.com/pytorch/executorch/pull/13924

Test Plan:
Imported from GitHub, without a `Test Plan:` line.

Rollback Plan:

Reviewed By: kimishpatel

Differential Revision: D81640227

Pulled By: metascroy
